homeassistant/components/touchlinesl/__init__.py

```diff
@@ -0,0 +1,55 @@
+"""The Roth Touchline SL integration."""
+
+from __future__ import annotations
+
+from pytouchlinesl import TouchlineSL
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+
+from .const import DOMAIN
+from .coordinator import TouchlineSLModuleCoordinator
+
+PLATFORMS: list[Platform] = [Platform.CLIMATE]
+
+type TouchlineSLConfigEntry = ConfigEntry[dict[str, TouchlineSLModuleCoordinator]]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: TouchlineSLConfigEntry) -> bool:
+    """Set up Roth Touchline SL from a config entry."""
+    account = TouchlineSL(
+        username=entry.data[CONF_USERNAME], password=entry.data[CONF_PASSWORD]
+    )
+
+    coordinators: dict[str, TouchlineSLModuleCoordinator] = {}
+    modules = await account.modules()
+
+    device_registry = dr.async_get(hass)
+    for module in modules:
+        coordinator = TouchlineSLModuleCoordinator(hass, module=module)
+        coordinators[module.id] = coordinator
+        await coordinator.async_config_entry_first_refresh()
```

Comment:
Rather use `asyncio.gather` here for the coordinator refresh (do a global search, some are doing this). Then loop over the coordinators for the device creation